src/types.jl

```diff
@@ -138,3 +138,9 @@ const ScalarType = Union{BitsType,Reference}
 
 # Define an H5O Object type
 const Object = Union{Group,Dataset,Datatype}
+
+idx_type(obj::Union{File,Group}) = get_track_order(get_create_properties(obj)) ? API.H5_INDEX_CRT_ORDER : API.H5_INDEX_NAME
```

Comment:
```suggestion
idx_type(obj::File) = get_context_property(:file_create).track_order ? API.H5_INDEX_CRT_ORDER : API.H5_INDEX_NAME
idx_type(obj::Group) = get_context_property(:group_create).track_order ? API.H5_INDEX_CRT_ORDER : API.H5_INDEX_NAME
```

Reply:
Perhaps we should always use `H5_INDEX_CRT_ORDER` if either the context or object property is enabled? Anyways, any fallback logic should be implemented here rather than `get_create_property` which has other uses.

```suggestion
idx_type(obj::File) =
    get_context_property(:file_create).track_order ||
    get_create_property(obj).track_order ? 
    API.H5_INDEX_CRT_ORDER : API.H5_INDEX_NAME
idx_type(obj::Group) =
    get_context_property(:group_create).track_order ||
    get_create_property(obj).track_order ? 
    API.H5_INDEX_CRT_ORDER : API.H5_INDEX_NAME
```